### PR TITLE
add clean targets

### PR DESCRIPTION
**Pull request by nhaghighat**
## Description

This PR adds clean targets to make to allow cleaner uninstall

## Checklist

- [ ] The PR branch is set up to merge into `feat/k8s` instead of `master`.

### CODE REVIEW

File: deploy/Makefile [hunk 1]
```diff
@@ -347,13 +399,41 @@ nfs-resources: ./bin/nfs-resources
...
+
+# Dummy clean target
+.PHONY: celan-
+clean-:
+	@echo "noting here fellas"
+
+# clean-all is meant to bring the cluster to its initial state, but currently it 
+# does not delete cert-manager and registry certificates, because the second 
+# cert-manager complains about unknown authority and can not create new certificates.
+.PHONY: celan-all
```
Comment (yannickperrenet) [inline deploy/Makefile:428]: Probably a typo: `celan-all` --> `clean-all`

File: deploy/Makefile [hunk 2]
```diff
@@ -347,13 +399,41 @@ nfs-resources: ./bin/nfs-resources
...
+
+.PHONY: clean-orchest-resources
+clean-orchest-resources:
+	helm uninstall orchest-resources --namespace ${NAMESPACE} --wait
+	rm ./bin/orchest-resources
+
+# Dummy clean target
+.PHONY: celan-
+clean-:
+	@echo "noting here fellas"
```
Comment (yannickperrenet) [inline deploy/Makefile:423]: I guess this can be removed.

File: deploy/Makefile [hunk 3]
```diff
@@ -278,6 +291,11 @@ jupyter-eg: ./bin/jupyter-eg
 	helm install ${JUPYTER_EG_DEPLOY_CONFIG}
 	touch $@
 
+.PHONY: clean-jupyter-eg
+clean-jupyter-eg:
+	helm uninstall jupyter-eg --namespace ${NAMESPACE} --wait
+	rm ./bin/jupyter-eg
```
Comment (yannickperrenet) [inline deploy/Makefile:297]: I don't think we deploy the jupyter-EG using helm right? So I guess the targets `clean-jupyter-eg` and `jupyter-eg` can be removed (just like the `deploy/thirdparty/enterprise-gateway` folder).

Comment (fruttasecca) [reply deploy/Makefile:297]: Indeed

File: deploy/Makefile [hunk 4]
```diff
@@ -269,6 +279,9 @@ node-agent:
 	${SHARED_DEPLOY_CONFIG} \
 	${NODE_AGENT_ENABLE_CONFIG} helm
 
+.PHONY: clean-node-agent
+clean-node-agent:
+	helm uninstall node-agent --namespace ${NAMESPACE} --wait
```
Comment (fruttasecca) [inline deploy/Makefile:281]: Isn't this cleaned already in `clean-orchest`? Or is the node-agent deployed separately?

File: deploy/Makefile (hunk 2 above)
Comment (nhaghighat) [reply deploy/Makefile]: No, this is here intentionally. once we deploy orchest with host-path enaled, the env var `STORAGE_CTL_TARGET` will be empty, and `make clean-all`, depends on this dummy clean target

File: deploy/Makefile (hunk 4 above)
Comment (nhaghighat) [reply deploy/Makefile:281]: the node-agent is usually deployed with orchest. but for debug purposes it has its own target (eventually all services might need a target for debug), so I also created a clean target just for that

Comment (fruttasecca) [reply deploy/Makefile:281]: I see, alright

**Review by fruttasecca — APPROVED** [review]
lgtm

**Pull request merged**